lib/gen_stage.ex

```diff
@@ -903,6 +921,12 @@ defmodule GenStage do
     end
   end
 
+  defp noreply_callback(:handle_demand, [demand, state], %{mod: mod, type: :producer_consumer} = stage) do
```

Comment:
Handling demand by the producer_consumer should be done by sending the demand to its producers upstream. :)